### PR TITLE
[Patch] Select fixes

### LINKED ISSUE #1248: Select event `onBlur` is executing when disabled
**Describe the bug**
The onBlur event is currently executing when the component is disabled. It also gains focus, leading to strange behaviour when tabbing between elements from the screen, since it does not jump between enabled items but also disabled selects.

**To Reproduce**

Code example:
```
const () => {
  const [value, setValue] = useState("");  
  const options = [
    { label: "Option 01", value: "1" },
    { label: "Option 02", value: "2" },
    { label: "Option 03", value: "3" },
    { label: "Option 04", value: "4" },
  ];
  const onBlur = ({ value, error }) => {
    setValue(value);
    console.log(value);
  };

  return (
    <DxcSelect
      label="Example"
      placeholder="Choose an option"
      options={options}
      value={value}
      onBlur={onBlur}
      disabled
    />
  );
};
```

**Expected behaviour**
The onBlur event must not be executed when the component is disabled.

### PR DESCRIPTION
**Pull request by GomezIvann**
Fixes #1248 and #1242. Also, the deprecated [`keycode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) has been updated to the `key` property for keyboard events in the Select component.

### CODE REVIEW

File: lib/src/select/Select.tsx [hunk 1]
```diff
@@ -289,17 +291,10 @@ const DxcSelect = React.forwardRef<RefType, SelectPropsType>(
       [handleSelectChangeValue, closeOptions, multiple]
     );
 
-    const handleListboxResize = () => {
+    const getSelectWidth = useCallback(() => {
       const rect = selectRef?.current?.getBoundingClientRect();
-      setListboxStyles({ width: rect?.width });
-    };
-    useEffect(() => {
```
Comment (Jialecl) [inline lib/src/select/Select.tsx:296]: remove useEffect import since it is no longer used.

Comment (GomezIvann) [reply lib/src/select/Select.tsx:296]: Done!

File: lib/src/select/Select.test.js [hunk 2]
```diff
@@ -293,26 +278,49 @@ describe("Select component tests", () => {
...
+    const { getByRole } = render(
+      <DxcSelect label="test-select-label" options={single_options} disabled onBlur={onBlur} />
+    );
+    const select = getByRole("combobox");
+
+    userEvent.click(select);
+    fireEvent.keyDown(getByRole("combobox"), { key: "Tab", code: "Tab", keyCode: 9, charCode: 9 });
+    expect(onBlur).not.toHaveBeenCalled();
+  });
+  test("Disabled select - When the component gains the focus, the listbox does not open", () => {
```
Comment (Jialecl) [inline lib/src/select/Select.test.js:314]: The select does not open when focused even when it's not disabled. We should test that the select can not gain the focus when disabled.
Also, the click test is repeated by the test in line 281.

Comment (GomezIvann) [reply lib/src/select/Select.test.js:314]: You are right Jiale! Already change it 👍🏼.

File: lib/src/select/Select.test.js [hunk 3]
```diff
@@ -293,26 +278,45 @@ describe("Select component tests", () => {
     expect(getByText("Option 02, Option 03, Option 04, Option 06")).toBeTruthy();
     expect(submitInput.value).toBe("4,2,6,3");
   });
-  test("Disabled select - Clear all options action must be shown but not clickable", () => {
-    const { getByRole, getByText, queryByRole } = render(
-      <DxcSelect label="test-select-label" value={["1", "2"]} options={single_options} disabled searchable multiple />
+  test("Disabled select - Listbox cannot be opened by click or keyboard press", () => {
```
Comment (Jialecl) [inline lib/src/select/Select.test.js:281]: Change the name of the test since there is no longer any keboard press.

Comment (GomezIvann) [reply lib/src/select/Select.test.js]: Done!

**Review by Jialecl — APPROVED** [review]

**Pull request merged**